From: Marc Kleine-Budde <m.kleine-budde@pengutronix.de>
Date: Wed, 25 Jan 2012 17:50:07 +0100
Subject: [PATCH] pppd: make makefile sysroot aware

The pppd Makefile is not SYSROOT aware. This patch replaces all
occurrences of absolute paths by their corresponding SYSROOT relative
ones.
Skip the 'wildcard' check completely. Libcrypt is always needed anyways.

Signed-off-by: Marc Kleine-Budde <m.kleine-budde@pengutronix.de>
Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 pppd/Makefile.linux | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/pppd/Makefile.linux b/pppd/Makefile.linux
index 4d536f84c7ba..30eb12f9d11d 100644
--- a/pppd/Makefile.linux
+++ b/pppd/Makefile.linux
@@ -100,8 +100,8 @@ endif
 
 # EAP SRP-SHA1
 ifdef USE_SRP
-CFLAGS	+= -DUSE_SRP -DOPENSSL -I/usr/local/ssl/include
-LIBS	+= -lsrp -L/usr/local/ssl/lib -lcrypto
+CFLAGS	+= -DUSE_SRP -DOPENSSL -I$(SYSROOT)/usr/local/ssl/include
+LIBS	+= -lsrp -L$(SYSROOT)/usr/local/ssl/lib -lcrypto
 TARGETS	+= srp-entry
 EXTRAINSTALL = $(INSTALL) -c -m 555 srp-entry $(BINDIR)/srp-entry
 MANPAGES += srp-entry.8
@@ -124,11 +124,9 @@ endif
 ifdef NO_CRYPT_HACK
 CFLAGS += -DNO_CRYPT_HACK
 else
-ifneq ($(wildcard /usr/include/crypt.h),)
 CFLAGS  += -DHAVE_CRYPT_H=1
 LIBS	+= -lcrypt
 endif
-endif
 
 ifdef USE_LIBUTIL
 CFLAGS	+= -DHAVE_LOGWTMP=1
@@ -137,7 +135,7 @@ endif
 
 ifdef NEEDDES
 ifndef USE_CRYPT
-CFLAGS   += -I/usr/include/openssl
+CFLAGS   += -I$(SYSROOT)/usr/include/openssl
 LIBS     += -lcrypto
 else
 CFLAGS   += -DUSE_CRYPT=1
